go/adbc/driver/bigquery/driver.go

```diff
@@ -0,0 +1,222 @@
+package bigquery
+
+import (
+	"cloud.google.com/go/bigquery"
+	"fmt"
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow-adbc/go/adbc/driver/internal/driverbase"
+	"github.com/apache/arrow/go/v16/arrow/memory"
+	"golang.org/x/exp/maps"
```

Comment:
are we actually using anything specific to `golang.org/x/exp/maps` that isn't in the updated standard maps package?

Reply:
Sorry it was similar to the snowflake implementation -- `maps.Clone` was used in `NewDatabase` so the `opts` is copied before passing to `db.SetOptions`. I wasn't sure if we need to copy that but I later decide to keep. I'll remove it.

Reply:
> Sorry it was similar to the snowflake implementation -- `maps.Clone` was used in `NewDatabase` so the `opts` is copied before passing to `db.SetOptions`. I wasn't sure if we need to copy that but I later decided to keep. I'll remove it.

Reply:
> Sorry it was similar to the snowflake implementation -- `maps.Clone` was used in `NewDatabase` so the `opts` is copied before passing to `db.SetOptions`. I wasn't sure if we need to copy that but I later decided to keep it. I'll remove it.